### ISSUE TITLE
Tests are running in parallel

### ISSUE THREAD

**Issue by AlexZeitler** [jprichardson/tape-promise#10]
I created two tests and it looks like they're running in parallel instead of in series.

These are my tests:

```js
const sleep = (ms = 0) => new Promise(fn => setTimeout(fn, ms));

test('Country Service should return Germany', async assert => {
  console.log('running test 1');
  await compose.kill({ cwd: __dirname });
  await compose.up({ cwd: path.join(__dirname) });
  const app = express();
  let server;

  app.post('/discovery', async (req, discoveryRes) => {
    discoveryRes.status(200).send();

    const res = await request({
      method: 'GET',
      uri: 'http://localhost:3002/countries',
      resolveWithFullResponse: true
    });

    server.on('close', () => {
      console.log('server 1 stopped');
      assert.equal('Germany', JSON.parse(res.body)[0]);
      assert.end();
    });

    server.close();
  });
  console.log('starting server 1');
  server = app.listen(5555);
});

test('Country should be created', async assert => {
  console.log(`starting test2`);
  console.log(`starting to sleep`);
  await sleep(5000);
  console.log(`slept long enough`);

  await compose.kill({ cwd: __dirname });
  await compose.up({ cwd: path.join(__dirname) });
  const app = express();
  let server;

  app.post('/discovery', async (req, discoveryRes) => {
    discoveryRes.status(200).send();

    const res = await request({
      method: 'POST',
      uri: 'http://localhost:3002/countries',
      resolveWithFullResponse: true,
      json: true,
      body: {
        name: 'Italy',
        isoalpha2: 'IT',
        isoalpha3: 'ITA',
        countrycode: '0039'
      }
    });

    server.on('close', () => {
      console.log('server 2 stopped');
      assert.equal(201, JSON.parse(res.statusCode));
      assert.end();
    });
    server.close();
  });

  console.log('starting server 2');
  server = app.listen(5555);
});

test('teardown', async assert => {
  assert.end();
});
```

This is the output:
```bash
TAP version 13
# Country Service should return Germany
running test 1
starting server 1
# Country should be created
starting test2
starting to sleep
server 1 stopped
ok 1 should be equal
slept long enough
starting server 2
# teardown
server 2 stopped
ok 2 should be equal

1..2
# tests 2
# pass  2

# ok
```

If I remove the `await sleep(5000);` call in the second test, the output looks like this

```bash
TAP version 13
# Country Service should return Germany
running test 1
starting server 1
# Country should be created
starting test2
starting to sleep
slept long enough
starting server 2
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::5555
    at Object.exports._errnoException (util.js:1014:11)
    at exports._exceptionWithHostPort (util.js:1037:20)
    at Server.setupListenHandle [as _listen2] (net.js:1307:14)
    at listenInCluster (net.js:1355:12)
    at Server.listen (net.js:1455:7)
```

**Comment by AlexZeitler** [comment]
When not using the `async` keyword for the `assert` but returning a `Promise` instead, everything works fine:

```js
test('Country Service should return Germany', assert =>
  new Promise(async (resolve, reject) => {
    console.log('running test 1');
    await compose.kill({ cwd: __dirname });
    await compose.up({ cwd: path.join(__dirname) });
    const app = express();
    let server;

    app.post('/discovery', async (req, discoveryRes) => {
      discoveryRes.status(200).send();

      const res = await request({
        method: 'GET',
        uri: 'http://localhost:3002/countries',
        resolveWithFullResponse: true
      });

      server.on('close', () => {
        console.log('server 1 stopped');
        assert.equal('Germany', JSON.parse(res.body)[0]);
        assert.end();

        return resolve();
      });

      server.close();
    });
    console.log('starting server 1');
    server = app.listen(5555);
  }));

test('Country should be created', assert =>
  new Promise(async (resolve, reject) => {
    console.log(`starting test2`);

    await compose.kill({ cwd: __dirname });
    await compose.up({ cwd: path.join(__dirname) });
    const app = express();
    let server;

    app.post('/discovery', async (req, discoveryRes) => {
      discoveryRes.status(200).send();

      const res = await request({
        method: 'POST',
        uri: 'http://localhost:3002/countries',
        resolveWithFullResponse: true,
        json: true,
        body: {
          name: 'Italy',
          isoalpha2: 'IT',
          isoalpha3: 'ITA',
          countrycode: '0039'
        }
      });

      server.on('close', () => {
        console.log('server 2 stopped');
        assert.equal(201, JSON.parse(res.statusCode));
        assert.end();
      });
      server.close();
    });
    console.log('starting server 2');
    server = app.listen(5555);
  }));

test('teardown', async assert => {
  // await compose.kill({ cwd: __dirname, log: true });

  assert.end();
});
```

**Comment by laat** [comment]
The tests in the first example can be reduced to something like this:

```js
test('Country Service should return Germany', async assert => {
  // ...
  server = app.listen(5555); // does not block, after this call the test is done.
});

test('Country should be created', async assert => {
  // ...
  server = app.listen(5555); // Should crash with EADDRINUSE until the first server is closed.
});
```

Basically, when an async test resolves, the next test should be run. Because the async function resolves before the server is closed, the next test is expected to fail.

Which makes the second example the correct way to write your test, because it resolves the promise after the server is closed.

**Comment by RyanZim** [comment]
This issue is terrible outdated, closing. Please open a new issue with a reduced test case if there's actually a bug with tape-promise.
